### PR TITLE
fix(invalidrole): allow upper and mixed case role names

### PR DESCRIPTION
**Pull request by lsprr**
Closes: #2695

### CODE REVIEW

**Comment by lsprr** [comment]
> Awesome! Thanks for the pr. We'll want to add a test to the [invalid role test file](https://github.com/dequelabs/axe-core/blob/develop/test/checks/shared/invalidrole.js) that verifies passing an uppercase role does not return `undefined`

Hi! Thank you for the feedback. I've added two tests to the [invalid role test file](https://github.com/dequelabs/axe-core/blob/develop/test/checks/shared/invalidrole.js) to ensure that passing an uppercase role to invalidroleEvaluate does not return undefined.

File: test/checks/shared/invalidrole.js [hunk 1]
```diff
@@ -111,4 +111,19 @@ describe('invalidrole', function () {
     );
     assert.deepEqual(checkContext._data, ['foo', 'bar']);
   });
+
+  it('should return true if applied to an uppercase nonsensical role', function () {
+    var virtualNode = queryFixture(
+      '<div id="target" role="FOO">Contents</div>'
```
Comment (straker) [inline test/checks/shared/invalidrole.js:117]: Thanks for the test. Sorry for the delay (been super busy with the 4.9 release happening soon). So an invalid role that is capitalized would still return true if the role was lowercased. We'll want a test that shows that a valid role that is uppercase will now return false. Something like:

```html
<div id="target" role="BUTTON">Contents</div>
```

Comment (lsprr) [reply test/checks/shared/invalidrole.js:117]: It's alright! Everyone is busy with their work/life, so no rush, :) I did add a test that shows that a valid role that is uppercase will now return false.

File: test/checks/shared/invalidrole.js [hunk 2]
```diff
@@ -111,4 +111,34 @@ describe('invalidrole', function () {
...
+    );
+    assert.isFalse(
+      checks.invalidrole.evaluate.call(
+        checkContext,
+        virtualNode.actualNode,
+        null,
+        virtualNode
+      )
+    );
+    assert.deepEqual(checkContext._data, ['BUTTON']);
```
Comment (straker) [inline test/checks/shared/invalidrole.js:142]: Thanks for the additional test. When the result is `false` we don't add a data property so this line needs to be removed (which is why the tests are failing).

```suggestion
```

**Review by straker — DISMISSED** [review]

**Pull request merged**